main.go

```diff
@@ -56,10 +33,68 @@ func main() {
 		push := payload.(github.PushPayload)
 		downloadURL := getHTTPDownloadURL(push)
 		Info.Println("Download URL:", downloadURL)
-		err = downloadRepo(downloadURL, "./test_repo")
+
+		// Get Repo Contents
+		err = downloadRepo(downloadURL, "./temp_repo")
 		if err != nil {
 			Error.Println("Error downloading repo:", err)
 		}
+
+		// Get random tar name
+		tar_name := guuid.New()
+
+		// Build image
+		Info.Println("Building image from Dockerfile...")
+		err = buildImageFromDockerfile(tar_name.String())
+		if err != nil {
+			Error.Println("Error building image from Dockerfile", err)
+		}
+
+		// Build tar
+		Info.Println("Building tar from Docker image...")
+		err = buildTarFromImage(tar_name.String())
+		if err != nil {
+			Error.Println("Error building tar from image", err)
+		}
+
+		tarNameExt := tar_name.String() + ".tar"
+		// Gzip tar
+		Info.Println("Gzipping tar...")
+		err = gzipTar(tarNameExt)
+		if err != nil {
+			Error.Println("Failure to gzip")
+			return
+		}
+		tarNameExt = tarNameExt + ".gz"
+
+		// Send .tar to worker
+		source := "/home/ubuntu/go/src/v9_deployment_manager/" + tarNameExt
+		destination := "/home/ubuntu/" + tarNameExt
+		err = scpToWorker(source, destination, tarNameExt)
+		if err != nil {
+			Error.Println("Error copying to worker", err)
+			return
+		}
+
+		// Activate worker
+		user := push.Repository.Owner.Login
+		repo := push.Repository.Name
+		dev := dev_id{user, repo, "test_hash"}
+		err = activateWorker(dev, "test", destination, tarNameExt)
```

Comment:
Do not hardcode the user/repo!

Reply:
This is not hard coded, this is reading the user and repo from the github push event